### ISSUE TITLE
🐛 Multi-line comments in Astro frontmatter can cause false positives

### ISSUE THREAD

**Issue by Boston343** [biomejs/biome#8882]
### Environment information

I am developing in Antigravity on Windows 11.

<details>

```bash

CLI:
  Version:                      2.3.12
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v24.11.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### What happened?

Issue I came across trying out biome in an astro project related to multi-line comments in the astro frontmatter.

If you have a multi line comment like the below, that contains something that could be a string opener (without a closer to match) then the rest of the file will show a bunch of errors:
```astro
---
/**
 * In this comment, if you add any string opening or closing, such as an apostrophy, the file will show 
 * a bunch of errors. Doesn't (remove the apostrophy in the previous word to fix) that stink? 
 */
import type { HTMLAttributes } from "astro/types";

type Props = HTMLAttributes<"div">;

const { class: className, ...rest } = Astro.props;
---

<div
  class="some-classes"
  data-state="closed"
  style="animation: none;"
  data-slot="accordion-content"
  {...rest}
>
  <div class="pt-0 pb-4">
    <slot />
  </div>
</div>

```

[playground demo](https://biomejs.dev/playground/?code=LQAtAC0ACgAvACoAKgAKACAAKgAgAEkAbgAgAHQAaABpAHMAIABmAGkAbABlACwAIABpAGYAIAB5AG8AdQAgAGEAZABkACAAYQBuAHkAIABzAHQAcgBpAG4AZwAgAG8AcABlAG4AaQBuAGcAIABvAHIAIABjAGwAbwBzAGkAbgBnACwAIABzAHUAYwBoACAAYQBzACAAYQBuACAAYQBwAG8AcwB0AHIAbwBwAGgAeQAsACAAdABoAGUAIABmAGkAbABlACAAdwBpAGwAbAAgAHMAaABvAHcAIAAKACAAKgAgAGEAIABiAHUAbgBjAGgAIABvAGYAIABlAHIAcgBvAHIAcwAuACAARABvAGUAcwBuACcAdAAgACgAcgBlAG0AbwB2AGUAIAB0AGgAZQAgAGEAcABvAHMAdAByAG8AcABoAHkAIABpAG4AIAB0AGgAZQAgAHAAcgBlAHYAaQBvAHUAcwAgAHcAbwByAGQAIAB0AG8AIABmAGkAeAApACAAdABoAGEAdAAgAHMAdABpAG4AawA%2FACAACgAgACoALwAKAGkAbQBwAG8AcgB0ACAAdAB5AHAAZQAgAHsAIABIAFQATQBMAEEAdAB0AHIAaQBiAHUAdABlAHMAIAB9ACAAZgByAG8AbQAgACIAYQBzAHQAcgBvAC8AdAB5AHAAZQBzACIAOwAKAAoAdAB5AHAAZQAgAFAAcgBvAHAAcwAgAD0AIABIAFQATQBMAEEAdAB0AHIAaQBiAHUAdABlAHMAPAAiAGQAaQB2ACIAPgA7AAoACgBjAG8AbgBzAHQAIAB7ACAAYwBsAGEAcwBzADoAIABjAGwAYQBzAHMATgBhAG0AZQAsACAALgAuAC4AcgBlAHMAdAAgAH0AIAA9ACAAQQBzAHQAcgBvAC4AcAByAG8AcABzADsACgAtAC0ALQAKAAoAPABkAGkAdgAKACAAIABjAGwAYQBzAHMAPQAiAHMAbwBtAGUALQBjAGwAYQBzAHMAZQBzACIACgAgACAAZABhAHQAYQAtAHMAdABhAHQAZQA9ACIAYwBsAG8AcwBlAGQAIgAKACAAIABzAHQAeQBsAGUAPQAiAGEAbgBpAG0AYQB0AGkAbwBuADoAIABuAG8AbgBlADsAIgAKACAAIABkAGEAdABhAC0AcwBsAG8AdAA9ACIAYQBjAGMAbwByAGQAaQBvAG4ALQBjAG8AbgB0AGUAbgB0ACIACgAgACAAewAuAC4ALgByAGUAcwB0AH0ACgA%2BAAoAIAAgADwAZABpAHYAIABjAGwAYQBzAHMAPQAiAHAAdAAtADAAIABwAGIALQA0ACIAPgAKACAAIAAgACAAPABzAGwAbwB0ACAALwA%2BAAoAIAAgADwALwBkAGkAdgA%2BAAoAPAAvAGQAaQB2AD4ACgA%3D&language=astro)

### Expected result

This should not cause errors, as anything in a comment should be valid and not interfere with later parsing.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
